### ISSUE TITLE
gson can not deserialize to immutable object if using staged builders

### ISSUE THREAD

**Issue by jvujcic** [immutables/immutables#558]
Hi,

following code will throw error:
**incompatible types: IdBuildStage cannot be converted to Builder
      ImmutableModel.Builder builder = ImmutableModel.builder();**

It looks like staged builders can't be used if we wont to deserialize to immutable object. If "stagedBuilder = true" is removed everything compiles fine.

```java
@Gson.TypeAdapters
@Value.Immutable
@Value.Style(get = { "get*", "is*" }, init = "set*", stagedBuilder = true)
public interface Model {
    public String getId();
}
```

```java
public class Main {

    public static void main(String[] args) {
        Gson gson = new GsonBuilder()
                .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_DASHES)
                .registerTypeAdapterFactory(new GsonAdaptersModel())
                .create();

        InputStream in = Main.class.getClassLoader().getResourceAsStream("test.json");

        try {
            Model test = gson.fromJson(new InputStreamReader(in), Model.class);
        }
        catch (JsonSyntaxException | JsonIOException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
    }
}
```

**Comment by lbialy** [comment]
Confirmed, generated code always uses general Builder type for variable created by `.builder()` call, which is a type mismatch with staged builder.

**Comment by elucash** [comment]
released in 2.4.4. thank you!
